packages/ember-template-compiler/lib/plugins/assert-no-each-in.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember-metal/core';
+import { assert } from 'ember-metal/debug';
+import calculateLocationDisplay from 'ember-template-compiler/system/calculate-location-display';
+
+function AssertNoEachIn(options = {}) {
+  this.syntax = null;
+  this.options = options;
+}
+
+AssertNoEachIn.prototype.transform = function AssertNoEachIn_transform(ast) {
+  if (!!Ember.ENV._ENABLE_LEGACY_VIEW_SUPPORT) {
```

Comment:
Not yet, we need a few changes to separate the `Ember.ENV` into its own module (possibly `ember-env`) first.